src/IssueViz.Security/Taint/TaintList/ViewModels/TaintIssuesControlViewModel.cs

```diff
@@ -177,9 +195,29 @@ private void UpdateIssues()
             NotifyPropertyChanged(nameof(HasServerIssues));
         }
 
+        private void UpdateCaption()
+        {
+            // We'll show the default caption if:
+            // * there are no underlying issues, or
+            // * there is not an active document.
+            // Otherwise, we'll add a suffix showing the number of issues in the active document.
+            string suffix = null;
+
+            if (unfilteredIssues.Count != 0 && activeDocumentFilePath != null)
+            {
+                suffix = $" ({GetFilteredIssuesCount()})";
+            }
+
+            WindowCaption = Resources.TaintToolWindowCaption + suffix;
+        }
+
+        private int GetFilteredIssuesCount() =>
+            IssuesView.OfType<object>().Count();
+
         private void Store_IssuesChanged(object sender, IssuesChangedEventArgs e)
         {
             UpdateIssues();
+            UpdateCaption();
```

Comment:
Minor: why not set the UpdateCaption inside UpdateIssues? This way we won't need to explicitly call it from the ctor as well.

Reply:
That's what I did originally but it's not enough on its own - the caption needs to be set after the issues have been set *and* the filter has been initialised.
In the end it was simpler just to make explicit calls in a couple of places.